Add modelInputs width, height, num_inference_steps, guidance_scale

gadicc opened the pull request:
# What is this?

Passes on some common model inputs (`width`, `height`, `num_inference_steps`, `guidance_scale`).

# Why?

They're useful :)

# How did you test it? 

Have been running locally in https://github.com/gadicc/stable-diffusion-react-nextjs-mui-pwa for a few days, and deployed.

# Comments

1. I have almost no Python experience, please comment on any styling / convention inconsistencies.
2. I used the defaults I found online, not sure if there's a better way to allow empty values and let the model itself use its defaults.
3. If this is all good I'll do the `init_image` and `strength` (iirc) stuff next when I get to that part of my app.

No urgency from my side, already running this deployed, but figured it would be useful for others.

gadicc committed:
Add modelInputs width, height, num_inference_steps, guidance_scale

PNRxA commented:
Argument positioning is wrong, Width changes the height. Needs a re-order or keyword arguments.

At least that's what happened when I tested locally

gadicc commented:
Ughhh thanks so much for reporting! :sweat_smile:   Have fixed this now.  Fixed the order AND switched to keyword arguments :sweat_smile: :sweat_smile:

0xspeedrunner commented:
@gadicc are there plans to merge this / are you using this in `prod` for Kiri?

gadicc commented:
@0xspeedrunner, here's the full source for what we're using prod for kiri.art :)

https://github.com/gadicc/banana-sd-base

It's not as beginner friendly as this repo but feel free to open any issues there if anything is unclear.
